version: '3.8'

services:
  results_flask_app:
    container_name: results_flask_app
    restart: always
    build:
      context: .
      dockerfile: flask_app/Dockerfile
    environment:
      - FLASK_APP=app.py
      - FLASK_ENV=production
      - FASTAPI_URL=http://results_fastapi_app:8000
      - DATABASE_URL=sqlite:////app/data/results.db
    volumes:
      - ./flask_app:/app
      - ./wait-for-fastapi.sh:/wait-for-fastapi.sh
      - shared_data:/app/data
    depends_on:
      - results_fastapi_app
    command: ["/wait-for-fastapi.sh", "python", "app.py"]
    networks:
      - amalfi_network
    expose:
      - 5000

  results_fastapi_app:
    container_name: results_fastapi_app
    restart: always
    build:
      context: .
      dockerfile: fastapi_app/Dockerfile
    environment:
      - FLASK_APP_URL=http://results_flask_app:5000/api/notify
      - DATABASE_URL=sqlite:////app/data/results.db
    volumes:
      - ./fastapi_app:/app
      - shared_data:/app/data
    networks:
      - amalfi_network
    expose:
      - 8000

networks:
  amalfi_network:
    external: true

volumes:
  shared_data:
    driver: local
